qr.go

```diff
@@ -0,0 +1,87 @@
+package txtdirect
+
+import (
+	"encoding/hex"
+	"fmt"
+	"image/color"
+	"net/http"
+	"strconv"
+
+	"github.com/mholt/caddy"
+	qrcode "github.com/skip2/go-qrcode"
+)
+
+// Qr contains Qr code generator's configuration
+type Qr struct {
+	Size            int
+	BackgroundColor string
+	ForegroundColor string
+
+	backgroundColor color.Color
+	foregroundColor color.Color
+}
+
+func (qr *Qr) Redirect(w http.ResponseWriter, r *http.Request) error {
+	Qr, err := qrcode.New(r.Host+r.URL.String(), qrcode.High)
+	if err != nil {
+		return fmt.Errorf("Couldn't generate the Qr instance: %s", err.Error())
+	}
+	if err = qr.ParseColors(); err != nil {
+		return fmt.Errorf("Coudln't parse colors: %s", err.Error())
+	}
+	Qr.BackgroundColor, Qr.ForegroundColor = qr.backgroundColor, qr.foregroundColor
+
+	Qr.Write(qr.Size, w)
```

Comment:
We should write a minimal html file instead. In Firefox for example a default image written back is transparent and therefore makes it hard to scan. Also helpful would be to add a clickable link to the actual webpage under the image so one can navigate there, if the qr code isn't necessary.

Reply:
I can see the reason to not add html. It would make for easier embedding.

Let's discuss this before implementing a change.

Reply:
Like the clickable link idea. But can't understand what you mean by easier embedding.

For the Firefox issue, we can add a custom white background color based on the request's `User Agent` header if writing HTML can't be considered.

Reply:
Any reason not to default to having a custom white background color in general?

Reply:
For context on the embedding.
Having the URL give back an image makes embedding into a different site etc. easier as it's just an image instead of playing with an iframe, when a full blown html would be given back. Does that make sense?

Reply:
I feel like dealing with iframe just makes the embedding harder. Think the image with a colored background should be enough and the user can add a link to the image while embedding it if they want it clickable.